share/eds/store.go

```diff
@@ -197,31 +201,59 @@ func (s *Store) Blockstore() bstore.Blockstore {
 	return s.bs
 }
 
-// CARBlockstore returns the IPFS blockstore that provides access to the IPLD blocks stored in an
-// individual CAR file.
-func (s *Store) CARBlockstore(ctx context.Context, dataHash []byte) (dagstore.ReadBlockstore, error) {
-	key := shard.KeyFromString(fmt.Sprintf("%X", dataHash))
+// CARBlockstore returns an IPFS Blockstore providing access to individual shares/nodes of a
+// specific EDS identified by DataHash and registered on the Store. NOTE: The Blockstore does not
+// store whole Celestia Blocks but IPFS blocks. We represent `shares` and NMT Merkle proofs as IPFS
+// blocks and IPLD nodes so Bitswap can access those.
+func (s *Store) CARBlockstore(
+	ctx context.Context,
+	root share.DataHash,
+) (dagstore.ReadBlockstore, error) {
+	key := shard.KeyFromString(root.String())
 	accessor, err := s.getAccessor(ctx, key)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("eds/store: failed to get accessor: %w", err)
 	}
-
 	return accessor.bs, nil
 }
 
-func (s *Store) getAccessor(ctx context.Context, key shard.Key) (*accessorWithBlockstore, error) {
-	// try to fetch from cache
-	accessor, err := s.cache.Get(key)
-	if err != nil && err != errCacheMiss {
-		log.Errorw("unexpected error while reading key from bs cache %s: %s", key, err)
+// GetDAH returns the DataAvailabilityHeader for the EDS identified by DataHash.
+func (s *Store) GetDAH(ctx context.Context, root share.DataHash) (*share.Root, error) {
+	key := shard.KeyFromString(root.String())
+	accessor, err := s.acquireShard(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("eds/store: failed to get accessor: %w", err)
 	}
-	if accessor != nil {
-		return accessor, nil
+	defer accessor.Close()
+
+	reader, err := carv1.NewCarReader(accessor)
```

Comment:
`carv1.ReadHeader` could be also an option here.

Reply:
forgot that existed, great catch

Reply:
done